dts/bindings/memory-controllers/atmel,sam-smc.yaml

```diff
@@ -0,0 +1,129 @@
+# Copyright (c) 2022, Basalte bv
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Atmel Static Memory Controller (SMC).
+
+  The SMC allows to interface with static-memory mapped external devices such as
+  SRAM, PSRAM, PROM, EPROM, EEPROM, LCD Module, NOR Flash and NAND Flash.
+
+  The SMC controller can have up to 4 children defining the connected external
+  memory devices. Device Tree example:
+
+  &smc {
+    status = "okay";
+    pinctrl-0 = <&smc_default>;
+    pinctrl-names = "default";
+
+    is66wv51216dbll@0 {
+      reg = <0>;
+
+      atmel,smc-write-mode = "nwe";
+      atmel,smc-read-mode = "nrd";
+      atmel,smc-setup-timing = <1 1 1 1>;
+      atmel,smc-pulse-timing = <6 6 6 6>;
+      atmel,smc-cycle-timing = <7 7>;
+    };
+  };
+
+  Finally, in order to make the memory available you will need to define new
+  memory device/s in DeviceTree:
+
+  sram1: sram@60000000 {
+      compatible = "zephyr,memory-region", "mmio-sram";
+      device_type = "memory";
+      reg = <0x60000000 DT_SIZE_K(512)>;
+      zephyr,memory-region = "SRAM1";
+  };
+
+compatible: "atmel,sam-smc"
+
+include: [base.yaml, pinctrl-device.yaml]
+
+properties:
+  reg:
+      required: true
+
+  label:
+    required: true
```

Comment:
Is the label really required?

Reply:
Not for me, I compared with other similar drivers, and it appears this is added for most.
Will remove.